crates/astria-sequencer-relayer/src/relayer/write/mod.rs

```diff
@@ -403,29 +417,62 @@ async fn submit_blobs(
     Ok(final_state)
 }
 
+async fn init_with_retry(client_builder: CelestiaClientBuilder) -> eyre::Result<CelestiaClient> {
+    let span = Span::current();
+
+    let retry_config = tryhard::RetryFutureConfig::new(u32::MAX)
+        .exponential_backoff(Duration::from_secs(1))
+        .max_delay(Duration::from_secs(30))
+        .on_retry(
+            |attempt: u32, next_delay: Option<Duration>, error: &CelestiaClientError| {
+                let wait_duration = next_delay
+                    .map(humantime::format_duration)
+                    .map(tracing::field::display);
+                warn!(
+                    parent: &span,
+                    attempt,
+                    wait_duration,
+                    error = %eyre::Report::new(error.clone()),
+                    "failed to initialize celestia client; retrying after backoff",
+                );
+                futures::future::ready(())
+            },
+        );
+
+    let celestia_client = tryhard::retry_fn(move || client_builder.clone().try_build())
+        .with_config(retry_config)
+        .in_current_span()
+        .await
+        .wrap_err("retry attempts exhausted; bailing")?;
+    info!("initialized celestia client");
+    Ok(celestia_client)
+}
+
 async fn submit_with_retry(
-    client: HttpClient,
+    client: CelestiaClient,
     blobs: Vec<Blob>,
     state: Arc<super::State>,
 ) -> eyre::Result<u64> {
-    use celestia_client::{
-        celestia_rpc::BlobClient as _,
-        celestia_types::blob::SubmitOptions,
-    };
     // Moving the span into `on_retry`, because tryhard spawns these in a tokio
     // task, losing the span.
     let span = Span::current();
+
+    // Create a watch channel to allow the `on_retry` function to provide the received
+    // `CelestiaClientError` to the next attempt of the `retry_fn`.
+    let (last_error_sender, last_error_receiver) = watch::channel(None);
```

Comment:
I'm updating to refetch cost params as requested, but this error channel is still useful, since the error could contain information about what fee we should retry using.